assets/course-theme/focus-mode.ts

```diff
@@ -63,12 +63,12 @@ window.addEventListener( 'DOMContentLoaded', () => {
 		.querySelector( '.sensei-course-theme__sidebar' )
 		?.addEventListener( 'transitionend', ( e ) => {
```

Comment:
```suggestion
		?.addEventListener( 'transitionend', ( e: TransitionEvent) => {
```
Can we remove the cast on the line 66 with this suggestion?

Reply:
Unfortunately, no. :(

TypeScript can be pretty annoying with event handlers used in Native DOM, because it assumes the event handlers are always of the `Event` type, and although `TransitionEvent` extends from `Event`, it complains that `Event` doesn't have the additional properties that `TransitionEvent` has.

This is fixed by the package `@types/web`, but, it apparently conflicts with `@types/node` (which we have installed because it is required by some of our own dependencies), and, given that we use mostly React on Sensei, I'm not sure if it's worth trying to fix it..

The experience on React is way better because React usually types each event handler individually, which makes the type-checking experience way better.